cla_backend/apps/call_centre/tests/api/test_case_api.py

```diff
@@ -285,6 +286,106 @@ def test_patch_provider_notes_not_allowed(self):
+        return self._test_assign_suggest(case, tz_model_mock, tz_helper_tz, category_assigned=False, out_of_hours=True)
+
+    @mock.patch("cla_provider.models.timezone.now")
+    @mock.patch("cla_provider.helpers.timezone.now")
+    def test_suggest_assign_case_with_category_out_of_hours(self, tz_model_mock, tz_helper_tz):
+        # should return null for suggested_provider and providers within the same category as suitable
+        case = make_recipe("legalaid.case", eligibility_check=self.cat_eligibility_check)
+        category = case.eligibility_check.category
+        self.assertEqual(category, self.suggested_category)
+        return self._test_assign_suggest(case, tz_model_mock, tz_helper_tz, category_assigned=True, out_of_hours=True)
+
+    @mock.patch("cla_provider.models.timezone.now")
+    @mock.patch("cla_provider.helpers.timezone.now")
+    def test_suggest_assign_case_with_category_in_hours(self, tz_model_mock, tz_helper_tz):
+        # return named provider for suggested_provider and  providers with the same category as suitable
+        case = make_recipe("legalaid.case", eligibility_check=self.cat_eligibility_check)
+        category = case.eligibility_check.category
+        self.assertEqual(category, self.suggested_category)
+        return self._test_assign_suggest(case, tz_model_mock, tz_helper_tz, category_assigned=True, out_of_hours=False)
+
+    def _test_assign_suggest(self, case, tz_model_mock, tz_helper_tz, category_assigned, out_of_hours):
+        # fix the date and time so we can check in and out of hours.
+        if out_of_hours:
+            fake_day = datetime.datetime(2014, 1, 2, 22, 1, 0).replace(tzinfo=timezone.get_current_timezone())
+        else:
+            fake_day = datetime.datetime(2014, 1, 2, 9, 1, 0).replace(tzinfo=timezone.get_current_timezone())
+
+        tz_model_mock.return_value = fake_day
+        tz_helper_tz.return_value = fake_day
+
+        url = reverse("call_centre:case-assign-suggest", args=(), kwargs={"reference": case.reference})
+        response = self.client.get(url, format="json", HTTP_AUTHORIZATION="Bearer %s" % self.token)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # check the suggested provider
+        if out_of_hours or not category_assigned:
+            self.assertIsNone(response.data["suggested_provider"])
+        else:
+            name = response.data["suggested_provider"]["name"]
+            self.assertIn(name, [unicode(x.name) for x in self.suggested_cat_providers])
+        #  now check the suitable_providers
+        if category_assigned:
+            suitable_providers = [ProviderSerializer(p).data for p in self.suggested_cat_providers]
+            self.assertEqual(response.data["suitable_providers"], suitable_providers)
+            # this should be all of the providers in suggested category
+
+            # self.assertItemsEqual(response.data["suitable_provider"], law_category=self.providers["category_id"]=])
```

Comment:
Do you need this commented out line of code?

Reply:
removed